app/controllers/application_controller/ci_processing.rb

```diff
@@ -388,6 +388,18 @@ def retirevms_now
 
   def check_compliance_vms
     assert_privileges(params[:pressed])
+
+    # Get all of the selected items if being in a list
+    records = find_records_with_rbac(record_class, checked_or_params)
+    # Check each item if there is any compliance policy assigned to it
+    if records.any? { |record| !record.has_compliance_policies? }
+      javascript_flash(
+        :text       => _('No Compliance Policies assigned to one or more of the selected items'),
+        :severity   => :error,
+        :scroll_top => true
+      )
```

Comment:
@hstastna My previous comment wasn't perfect it was missing a `return`.

Reply:
I know that there's no `return`. Your previous comment gave me idea if it is really necessary.

Reply:
Well it broke your specs. And this way we can get error and success messages at the same time if one selected Vm is ok and other's not.

Reply:
I've made the changes. Now it should be ok.